plugins/exec/keys.ml

```diff
@@ -0,0 +1,15 @@
+type t = Cons of (int * (unit -> t))
```

Comment:
What does motivate this complexity?

Reply:
Being able to reuse “released” keys, basically. AFAICT, the `Stream` module of OCaml does not provide a way to push back an item in the stream.

Note that reusing released keys is not really mandatory per se, but could help uncovering bugs while writing impure programs (see the first commit messages, where I mention use-after-free issues).

Reply:
After a short but insightful discussion with @mbty-ssi I will probably drop the idea to reuse keys; at the very least, use-after-free without reused keys will provoke the raise of a `Not_found`-like exception, and this is more than enough for finding bugs.